src/ocean/util/container/AppendBuffer.d

```diff
@@ -549,12 +551,9 @@ public class AppendBuffer ( T, Base: AppendBufferImpl ): Base, IAppendBufferRead
          **********************************************************************/
 
         T cut ( )
-        in
-        {
-            assert (this.length, "cannot cut last element: content is empty");
-        }
-        body
         {
+            verify (this.length > 0, "cannot cut last element: content is empty");
```

Comment:
Only thing that was scaring me while I was doing the same is that, if the argument is signed (granted, it's not in this case), _and_ negative value is expected (think returning negative value on error, positive on success), this will be a change in the semantics. Again, I don't think it matters in this (or most (all?) cases), but we should be careful.

Reply:
I was going to start doing `verify(!!this.length)`, but I opted out :-)

Reply:
Yeah, right. I tried to check them, but it's worth double-checking.